Aplib.Core/Desire/DesireSets/DesireSet.cs

```diff
@@ -2,12 +2,13 @@
 using Aplib.Core.Collections;
 using Aplib.Core.Desire.Goals;
 using Aplib.Core.Desire.GoalStructures;
+using System.Collections.Generic;
 using System.Linq;
 
 namespace Aplib.Core.Desire.DesireSets
 {
     /// <inheritdoc cref="DesireSet{TBeliefSet}"/>
-    public class DesireSet<TBeliefSet> : IDesireSet<TBeliefSet>, IDocumented
+    public class DesireSet<TBeliefSet> : IDesireSet<TBeliefSet>, ILoggable
```

Comment:
Why is it not IDocumented anymore? It still has our Metadata. I opt for renaming it to IHasMetadata for clarity.

Also, can the _interfaces_ be ILoggable instead? That way your logger can be used for any user-implemented DesireSets/GoalStructures/Goals/Tactics/Actions as well.

Reply:
By making the interfaces ILoggable you also don't need to do all the checks for `is ILoggable`

Reply:
We had a discussion about this previously previously that was not completely resolved, but the counter-argument is that you generally don't want interfaces to be deriving other interfaces. They are supposed to be mostly independent, if possible.

Reply:
Yeah in the end I chose to put it on the implementations. That way you can still make a different implementation that is not loggable if you want to.